xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm

```diff
@@ -0,0 +1,136 @@
+/*
+ *  Copyright (C) 2010-2019 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#import "DarwinEmbedNowPlayingInfoManager.h"
+
+#import "platform/darwin/ios/XBMCController.h"
+
+#import <Foundation/Foundation.h>
+#import <MediaPlayer/MPMediaItem.h>
+#import <MediaPlayer/MPNowPlayingInfoCenter.h>
+
+@implementation DarwinEmbedNowPlayingInfoManager
+
+@synthesize playbackState;
+
+#pragma mark - Now Playing routines
+
+- (void)setDarwinEmbedNowPlayingInfo:(NSDictionary*)info
+{
+  self.nowPlayingInfo = info;
+  [MPNowPlayingInfoCenter defaultCenter].nowPlayingInfo = self.nowPlayingInfo;
+}
+
+- (void)onPlay:(NSDictionary*)item
+{
+  NSMutableDictionary* dict = [[NSMutableDictionary alloc] init];
+
+  NSString* title = item[@"title"];
+  if (title && title.length > 0)
+    dict[MPMediaItemPropertyTitle] = title;
+  NSString* album = item[@"album"];
+  if (album && album.length > 0)
+    dict[MPMediaItemPropertyAlbumTitle] = album;
+  NSArray* artists = item[@"artist"];
+  if (artists && artists.count > 0)
+    dict[MPMediaItemPropertyArtist] = [artists componentsJoinedByString:@" "];
+  if (NSNumber* track = item[@"track"])
+    dict[MPMediaItemPropertyAlbumTrackNumber] = track;
+  if (NSNumber* duration = item[@"duration"])
+    dict[MPMediaItemPropertyPlaybackDuration] = duration;
+  NSArray* genres = item[@"genre"];
+  if (genres && genres.count > 0)
+    dict[MPMediaItemPropertyGenre] = [genres componentsJoinedByString:@" "];
+
+  NSString* thumb = [item objectForKey:@"thumb"];
+  if (thumb && thumb.length > 0)
+  {
+    auto image = [UIImage imageWithContentsOfFile:thumb];
+    if (image)
+    {
+      MPMediaItemArtwork* mArt = [[MPMediaItemArtwork alloc] initWithBoundsSize:image.size requestHandler:^UIImage* _Nonnull(CGSize aSize) { return image; }];
+      if (mArt)
+        dict[MPMediaItemPropertyArtwork] = mArt;
+    }
+  }
+
+  if (NSNumber* elapsed = item[@"elapsed"])
+    dict[MPNowPlayingInfoPropertyElapsedPlaybackTime] = elapsed;
+  if (NSNumber* speed = item[@"speed"])
+    dict[MPNowPlayingInfoPropertyPlaybackRate] = speed;
+  if (NSNumber* current = item[@"current"])
+    dict[MPNowPlayingInfoPropertyPlaybackQueueIndex] = current;
+  if (NSNumber* total = item[@"total"])
+    dict[MPNowPlayingInfoPropertyPlaybackQueueCount] = total;
+
+  /*! @Todo additional properties?
+   other properities can be set:
+   MPMediaItemPropertyAlbumTrackCount
+   MPMediaItemPropertyComposer
+   MPMediaItemPropertyDiscCount
+   MPMediaItemPropertyDiscNumber
+   MPMediaItemPropertyPersistentID
+   Additional metadata properties:
+   MPNowPlayingInfoPropertyChapterNumber;
+   MPNowPlayingInfoPropertyChapterCount;
+   */
+
+  [self setDarwinEmbedNowPlayingInfo:dict];
+
+  self.playbackState = DARWINEMBED_PLAYBACK_PLAYING;
+
+#if defined(TARGET_DARWIN_IOS)
```

Comment:
might make more sense to subclass on iOS and perform additional logic in the subclass

Reply:
As this is IOS only PR, i'll drop that ifdef completely for now. Can deal with how to handle it when tvos PR comes through.